std.go

```diff
@@ -34,25 +38,61 @@ func (StdRoller) Roll(matches []string) (RollResult, error) {
 		return nil, err
 	}
 
+	keep := ""
+	num := 0
+	if matches[4] != "" {
+		number, err := strconv.ParseInt(matches[5], 10, 0)
+		if err != nil {
+			return nil, err
+		}
+		num = int(number)
+		keep = matches[4]
+	}
+	
 	result := StdResult{
 		basicRollResult: basicRollResult{matches[0]},
 		Rolls:           make([]int, dice),
+		Dropped:		 nil,
 		Total:           0,
 	}
 
-	if matches[3] != "" {
-		bonus, err := strconv.ParseInt(matches[3], 10, 0)
+	if matches[6] != "" {
+		bonus, err := strconv.ParseInt(matches[6], 10, 0)
 		if err != nil {
 			return nil, err
 		}
 		result.Total += int(bonus)
 	}
 
-	for i := int64(0); i < dice; i++ {
+	for i := 0; i < len(result.Rolls); i++ {
 		roll := rand.Intn(int(sides)) + 1
-		result.Total += roll
 		result.Rolls[i] = roll
 	}
+
+	sort.Ints( result.Rolls )
+	size := len( result.Rolls )
+	
+	if keep == "kh" {
+		result.Dropped = result.Rolls[:size-num]
+		result.Rolls = result.Rolls[size-num:]
+	}
```

Comment:
Not a huge deal, but you could make this a `switch` statement, which might look cleaner. I'm fine with it either way though, to be honest.